src/lib/DutchDecayLib.sol

```diff
@@ -68,6 +68,35 @@ library DutchDecayLib {
         }
     }
 
+    /// @notice returns the linear interpolation between the two points
+    /// @param startPoint The start of the decay
+    /// @param endPoint The end of the decay
+    /// @param currentPoint The current position in the decay
+    /// @param startAmount The amount of the start of the decay
+    /// @param endAmount The amount of the end of the decay
+    function linearDecay(
```

Comment:
I added another version of this that accepts int256 for start and end amount.
1. I wanted to decay only through the relative amounts so that I can apply the bounding logic (min/max) check after the decay.
2. It was too convoluted to try to support both in the same function. 

Note that this linearDecay performs `mulDivDown` in both positive and negative cases as it felt more intuitive. Happy to discuss!

Reply:
can the uint one call this one w/ casted params?

Reply:
Yup! I was trying to do the other way around, but as long as we're okay with losing 1 bit of precision, it's cleaner.